squeal-postgresql/src/Squeal/PostgreSQL/Definition.hs

```diff
@@ -79,3 +85,21 @@ manipDefinition
   -- ^ no input or output
   -> Definition schemas schemas
 manipDefinition = UnsafeDefinition . (<> ";") . renderSQL
+
+-- | Like @PGTyped@ but also accounts for nullity.
+class PGNullityTyped schemas (nullty :: NullityType) where
+  pgNullityType :: ColumnTypeExpression schemas ('NoDef :=> nullty)
+
+instance PGTyped schemas ('Null ty) => PGNullityTyped schemas ('Null ty) where
+  pgNullityType = nullable (pgtype @_ @('Null ty))
+
+instance PGTyped schemas ('NotNull ty) => PGNullityTyped schemas ('NotNull ty) where
+  pgNullityType = notNullable (pgtype @_ @('NotNull ty))
+
+-- | Allow you to specify pg column types in relation to haskell types.
+-- | >>> hask @(Maybe String) ~ nullable text
```

Comment:
is this being doctested? might be nice to see the `printSQL` output.

Reply:
I'm not familiar with how to set that up.

Reply:
I think if you just change `-- | >>>` to `-- >>>` it should be able run doctests on your examples. To run just the doctests with stack, run
```
stack test squeal-postgresql:doctest
```

Reply:
You can think of each `>>>` as a command in GHCi

Reply:
It doesn't work "as is" (with removing `|`). It might require enabling some extensions and some extra boilerplate to transform it into a proper expression. I don't have time to get into this right now. Might do it later.

Reply:
likely something like
```haskell
-- >>> printSQL $ hask @(Maybe String)
-- text NULL
--
-- >>> printSQL $ hask @Int32
-- int8 NOT NULL
```

Reply:
That seems to work.

Thanks.